operator-metadata/automation/modules/bundle_automation.py

```diff
@@ -101,10 +101,22 @@ def get_nvr(brew_client, prefix, version):
         raise ValueError('No NVR found in brew with prefix %s and version %s' % (prefix, version))
 
     @staticmethod
-    def get_pull_spec_from_info(info):
+    def get_digest_from_info(info):
         data = yaml.safe_load(info)
         return data["extra"]["image"]["index"]["digests"]["application/vnd.docker.distribution.manifest.list.v2+json"]
 
+    @staticmethod
+    def get_tag_from_info(info):
+        data = yaml.safe_load(info)
+        return data["extra"]["image"]["index"]["tags"][0]
+
+    @staticmethod
+    def get_image_name_from_info(info):
+        data = yaml.safe_load(info)
+        pull_spec = data["extra"]["image"]["index"]["pull"][0]
+        image_name = pull_spec.split("/")[2].split("@")[0]
+        return image_name
+
```

Comment:
The full pull spec using the NVR is available in the second array entry, offset 1. No need to split from SHA and recombine with the tag.

```suggestion
    @staticmethod
    def get_pull_spec_from_info(info):
        data = yaml.safe_load(info)
        pull_spec = data["extra"]["image"]["index"]["pull"][1]
        return pull_spec

```

Reply:
From what I understand, `data["extra"]["image"]["index"]["pull"][1]` contains pull specs in the OSBS pullspec format:
```
<PLACEHOLDER>/rh-osbs/amq-streams-bridge-rhel8:2.5.0-5
```
It doesn't use the same registry, namespace, or image name that are used in the CSV. 

The code extracts the image name from these pull_specs to get the image name that is used in the CSV e.g. `bridge-rhel8`. Then uses this image name along with corresponding tag to search and replace the old image tags with the new ones.

 Just updated the code to make this more clear using a mix of the suggestion above, let me know if this is more clear

Reply:
> It doesn't use the same registry, namespace, or image name that are used in the CSV.

That is correct. However, the magic of the configurations under `operator_manifests` in the `container.yaml` file allow OSBS to modify the pull specs to reflect the final repository, namespace, and name.

Reply:
Discussed this with Mike on a call, because this CSV is public on GitHub it may be more prudent to stick to using external registry/pull spec addresses here. If we do this, we should stick with this automation change for now and update it at a later date once we move this source to internal repos

Reply:
Is that a good summary @MikeEdgar?